client/src/main/java/com/battleships/connection/commands/ServerCommandsFactory.java

```diff
@@ -0,0 +1,31 @@
+package com.battleships.connection.commands;
+
+import com.battleships.commands.CommandType;
+import com.battleships.commands.Message;
+import com.battleships.connection.commands.server.commands.*;
+
+public class ServerCommandsFactory {
+
+    public static <V> AbstractServerCommand getCommandImpl(Message<V> message) {
+        CommandType commandType = message.getCommandType();
+        V value = message.getValue();
+        switch (commandType) { // TODO Krzysiek 16.07 replace switch with sth better
+            case PLAYER_REGISTERED_SUCCESSFULLY:
+                return new PlayerRegistered<>(value);
+            case START_PLAYING:
```

Comment:
In the future please consider using polimorphism instead of switch case instruction.

Reply:
Thanks for the comment! We are thinking of implementing a strategy pattern here